foundation/foundation-mda/src/main/resources/templates/data-delivery-data-records/spark.schema.base.java.vm

```diff
@@ -90,30 +92,29 @@ public abstract class ${record.capitalizedName}SchemaBase extends SparkSchema {
     public Dataset<Row> validateDataFrame(Dataset<Row> data) {
         Dataset<Row> dataWithValidations = data
             #foreach ($field in $record.fields)
-                #set ( $columnName = "#if($field.column)$field.column#{else}$field.upperSnakecaseName#end" )
                 #if (${field.isRequired()})
-                    .withColumn("${columnName}_IS_NOT_NULL", col(${columnName}_COLUMN).isNotNull())
```

Comment:
I: This is the original bug. When `field.column` is set to some name, the constant is referenced here as `col(my_col_name_COLUMN)` even though the constant is still `MY_FIELD_COLUMN`. To help ensure this mistake doesn't happen again, we just create a mapping of field to constant at the top of the file and use that mapping any time we want to reference the constant.